lib/ansible/modules/system/systemd-modules-load.py

```diff
@@ -0,0 +1,147 @@
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: systemd-modules-load
+short_description:  systemd-modules-load
+description:
+    - Allows you to  generate systemd-modules-load configuration files.
+version_added: "2.8"
+options:
+    conf_path:
+        description:
+            - Specifies the path where to write the configuration files.
+        default: "/etc/modules-load.d"
+        choices: [ "/etc/modules-load.d", "/run/modules-load.d", "/usr/lib/modules-load.d" ]
+    file_name:
+        description:
+            - This configuration file name where the configurations will be written. Note the file name will be
+              automatically have the extension .conf.
+    modules:
+        description:
+            - Space separated list of module names.
+    state:
+        description:
+            - Whether configuration files should be added or removed.
+        choices: [ "create", "remove" ]
+author: "Susant Sahani (@ssahani) <susant@redhat.com>"
+'''
+
+EXAMPLES = '''
+# Create config file
+- systemd-modules-load:
+       file_name: my_tunnel
+       modules:  ipip sit vti
+       state: create
+
+# Remove config file
+- systemd-modules-load:
+       file_name: ipip
+       state: remove
+'''
+
+RETURN = r'''
+'''
+
+import os
+from ansible.module_utils.basic import get_platform, AnsibleModule
+
+UNIT_PATH_MODULES_LOAD = '/usr/lib/modules-load.d'
+UNIT_PATH_MODULES_LOAD_SYSTEM = '/etc/modules-load.d'
+UNIT_PATH_MODULES_LOAD_RUN = '/run/modules-load.d'
+
+
+class SystemdModulesLoad(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.args = self.module.params
+        self.conf_path = module.params['conf_path']
+        self.file_name = module.params['file_name']
+        self.modules = module.params['modules']
+        self.state = module.params['state']
+        self.changed = False
+
+    def remove_files(self):
+        paths = [UNIT_PATH_MODULES_LOAD_RUN, UNIT_PATH_MODULES_LOAD_SYSTEM, UNIT_PATH_MODULES_LOAD]
+        rc = False
+
+        list_conf_files = self.file_name.split(' ')
+        for conf_file in list_conf_files:
+            conf_file += '.conf'
+            if os.path.exists(os.path.join(self.conf_path, conf_file)):
+                os.remove(os.path.join(self.conf_path, conf_file))
+                rc = True
+
+        return rc
+
+    def write_configs_to_file(self):
+        if not os.path.exists(self.conf_path):
+            os.makedirs(self.conf_path, exist_ok=True)
+
+        self.file_name += '.conf'
+        with open(os.path.join(self.conf_path, self.file_name), "w") as f:
+            modules_list = self.modules.split(' ')
+            for kmod in modules_list:
+                f.write(kmod + '\n')
+            f.close()
+            return True
+
+        return False
+
+    def configure_modules_load(self):
```

Comment:
this forcibly creates/removes the files w/o checking if they exist or not, doing so would allow you to offer a 'changed' s state. Also it is encouraged that modules are safe to re-run w/o affecting systems when not needed.

Reply:
I hope I followed you comment right.